### PR TITLE
Clear validation messages when form reset

### PR DESCRIPTION
**Pull request by slavko-ptitsyn**
Possible solution for the problem in #93

### CODE REVIEW

**Comment by slavko-ptitsyn** [comment]
I've just signed it

File: iron-form.html [hunk 1]
```diff
@@ -404,6 +404,7 @@
           } else {
             el.value = this._customElementsInitialValues[i];
           }
+          el.invalid = false;
```
Comment (notwaldorf) [inline iron-form.html]: Would you mind adding a test for this case, please?

**Comment by slavko-ptitsyn** [comment]
@notwaldorf I added tests

**Comment by notwaldorf** [comment]
Awesome! Thanks so much!

**Pull request merged**